pkg/filter/pipelinerun.go

```diff
@@ -0,0 +1,142 @@
+package filter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/triggers/test/stubs"
+
+	tknv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"knative.dev/pkg/apis"
+)
+
+// LabelKeyPrefix prefix used in all labels.
+const LabelKeyPrefix = "trigger.shipwright.io"
+
+var (
+	// OwnedByRunLabelKey labels the BuildRun as owned by Tekton Run.
+	OwnedByRunLabelKey = fmt.Sprintf("%s/owned-by-run", LabelKeyPrefix)
+	// OwnedByPipelineRunLabelKey lables the BuildRun as owned by Tekton PipelineRun.
+	OwnedByPipelineRunLabelKey = fmt.Sprintf("%s/owned-by-pipelinerun", LabelKeyPrefix)
+	// BuildRunsCreatedKey labels the PipelineRun with the BuildRuns created.
+	BuildRunsCreatedKey = fmt.Sprintf("%s/buildrun-names", LabelKeyPrefix)
+	// PipelineRunNameKey labels PipelineRuns with its current name, to avoid object reprocessing.
+	PipelineRunNameKey = fmt.Sprintf("%s/pipelinerun-name", LabelKeyPrefix)
+)
+
+// searchBuildRunForRunOwner inspect the object owners for Tekton Run and returns it, otherwise nil.
+func searchBuildRunForRunOwner(br *v1alpha1.BuildRun) *types.NamespacedName {
+	for _, ownerRef := range br.OwnerReferences {
+		if ownerRef.APIVersion == stubs.TektonAPIv1alpha1 && ownerRef.Kind == "Run" {
```

Comment:
Something to keep note of - I believe more recent versions of Tekton have a beat version of the `Run` object.

This should be filed as an issue.

Reply:
Sure, that's a good point, Adam. Currently the `Run` resource is still in [`v1alpha1` version](https://github.com/tektoncd/pipeline/blob/f6f9f3eb2109f8221dacf7eaf0d7cd3f919654a1/pkg/client/listers/pipeline/v1alpha1/run.go#L19), and a new issue has been filed (#7).

Reply:
> I think there are only two things that need to be changed here - the domain, and that bit about co-mingling test code. Otherwise let's merge this and get the project rolling!

Thanks for the review! Please consider the latest changes 👍